generators/app/index.js

```diff
@@ -336,6 +337,29 @@ module.exports = JhipsterGenerator.extend({
     },
 
     end: {
+        localInstall: function() {
+            if (this.skipClient === true) {
+                // Generate a package.json file containing the current version of the generator as dependency
+                this.template('_skipClientApp.package.json', 'package.json', this, {});
+
+                var done = this.async();
+                var errorLocalJhipsterInstall = function (err) {
+                    if (err) {
+                        this.warning('Could not install generator-jhipster locally.');
+                    }
+                    done();
+                };
+                if(this.yarnInstall) {
+                    this.log(chalk.bold('\nInstalling generator-jhipster locally using yarn'));
+                    shelljs.exec('yarn install', errorLocalJhipsterInstall);
```

Comment:
please use `spawnCommand` method from yeoman for yarn and `npmInstall` method for npm  http://yeoman.io/authoring/dependencies.html